platform/macos/godot_application_delegate.mm

```diff
@@ -113,6 +106,37 @@ - (void)handleAppleEvent:(NSAppleEventDescriptor *)event withReplyEvent:(NSApple
 	}
 }
 
+- (void)handleAppleEvent:(NSAppleEventDescriptor *)event onAppUrlEvent:(NSAppleEventDescriptor *)replyEvent {
+	OS_MacOS *os = (OS_MacOS *)OS::get_singleton();
+	if (!event || !os) {
+		return;
+	}
+
+	List<String> args;
```

Comment:
`List<String>` is not really needed, unlike file open event, URL is always a single string, so it can go directly to `str`.

Reply:
on it